### PR TITLE
Improved DAV synchronization

### PR DESCRIPTION
**Pull request by arek1-618**
firefox_2019-03-07_13-29-55
firefox_2019-03-07_13-17-03
firefox_2019-03-07_12-41-47

### CODE REVIEW

File: layouts/basic/modules/Calendar/Detail/BlockView.tpl [hunk 1]
```diff
@@ -23,7 +23,7 @@
 								{assign var=TITLE value=''}
 								{if $INVITIE['crmid']}
 									{assign var=INVITIE_RECORD value=vtlib\Functions::getCRMRecordMetadata($INVITIE['crmid'])}
-									{assign var=LABEL value=$INVITIE_RECORD['label']}
+									{assign var=LABEL value=vtlib\Functions::getCRMRecordLabel($INVITIE['crmid'])}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/Detail/BlockView.tpl]: ?

File: layouts/basic/modules/Calendar/Detail/BlockView.tpl [hunk 2]
```diff
@@ -23,7 +23,7 @@
 								{assign var=TITLE value=''}
 								{if $INVITIE['crmid']}
 									{assign var=INVITIE_RECORD value=vtlib\Functions::getCRMRecordMetadata($INVITIE['crmid'])}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/Detail/BlockView.tpl]: ?

File: layouts/basic/modules/Calendar/InviteRow.tpl [hunk 3]
```diff
@@ -10,11 +11,12 @@
 				<span class="input-group-text">
 					{if $INVITIE['crmid']}
 						{assign var=INVITIE_RECORD value=vtlib\Functions::getCRMRecordMetadata($INVITIE['crmid'])}
-						{assign var=LABEL value=$INVITIE_RECORD['label']}
+						{assign var=LABEL value=vtlib\Functions::getCRMRecordLabel($INVITIE['crmid'])}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/InviteRow.tpl]: ?

File: layouts/basic/modules/Calendar/InviteRow.tpl [hunk 4]
```diff
@@ -10,11 +11,12 @@
 				<span class="input-group-text">
 					{if $INVITIE['crmid']}
 						{assign var=INVITIE_RECORD value=vtlib\Functions::getCRMRecordMetadata($INVITIE['crmid'])}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/InviteRow.tpl]: ?

File: modules/API/models/CalDAV.php [hunk 5]
```diff
@@ -272,15 +272,17 @@ public function recordSync()
 	public function recordCreate($cal)
 	{
 		\App\Log::trace(__METHOD__ . ' | Start Cal ID' . $cal['id']);
+		\App\DebugerEx::log('recordCreate', $cal['calendardata']);
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CalDAV.php]: ?

File: modules/Calendar/models/Record.php [hunk 6]
```diff
@@ -80,6 +81,7 @@ public function getName()
 		if (empty($name)) {
 			$name = parent::getName();
 		}
+
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/models/Record.php]: ?

File: modules/Calendar/models/Record.php [hunk 7]
```diff
@@ -37,6 +37,7 @@ public static function getNameByReference($refModuleName)
 			$fieldName = $relationModel->getRelationField()->getFieldName();
 			Vtiger_Cache::set('NameRelatedField', $refModuleName . '-Calendar', $fieldName);
 		}
+
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/models/Record.php]: ?

File: modules/Calendar/models/Record.php [hunk 8]
```diff
@@ -248,21 +250,27 @@ public function updateReminderPostpone(string $time)
...
 				$datatime = date('Y-m-d H:i:s', strtotime('+120 min'));
+
 				break;
 			case '6h':
 				$datatime = date('Y-m-d H:i:s', strtotime('+6 hour'));
+
 				break;
 			case '1d':
 				$datatime = date('Y-m-d H:i:s', strtotime('+1 day'));
+
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/models/Record.php]: ?

File: app/Fields/Phone.php [hunk 9]
```diff
@@ -18,11 +18,11 @@ class Phone
...
 	 *
 	 * @param string      $phoneNumber
-	 * @param string|null $phoneCountry
+	 * @param null|string $phoneCountry
 	 *
-	 * @return bool|array
+	 * @return array|bool
 	 */
-	public static function getDetails(string $phoneNumber, ?string $phoneCountry = null)
+	public static function getDetails(?string $phoneNumber, ?string $phoneCountry = null)
```
Comment (mariuszkrzaczkowski) [inline app/Fields/Phone.php]: ?

File: layouts/basic/modules/Calendar/EditViewBlocks.tpl [hunk 10]
```diff
@@ -5,7 +5,7 @@
 	{include file=\App\Layout::getTemplatePath('EditViewBlocks.tpl', 'Vtiger')}
 	<input type="hidden" name="userChangedEndDateTime" value="{$USER_CHANGED_END_DATE_TIME}"/>
 	{assign var=IS_HIDDEN value=false}
-	{assign var=INVITIES_SELECTED value=$RECORD->getInvities()}
+	{assign var=INVITIES_SELECTED value=$RECORD->getInvitiesWithCRM()}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/EditViewBlocks.tpl]:
```suggestion
	{assign var=INVITIES_SELECTED value=$RECORD->getInvitiesWithCrm()}
```

File: modules/API/models/CalDAV.php [hunk 11]
```diff
@@ -596,10 +593,10 @@ protected function davSaveAttendee(array $record, Sabre\VObject\Component\VCalen
 		}
 		foreach ($invities as &$row) {
 			$attendee = $vcalendar->createProperty('ATTENDEE', 'mailto:' . $row['email']);
-			$attendee->add('CN', vtlib\Functions::getCRMRecordLabel($row['id']));
+			$attendee->add('CN', vtlib\Functions::getCRMRecordLabel($row['crmid']));
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CalDAV.php]: App\Record::getLabel

File: modules/API/models/CalDAV.php [hunk 12]
```diff
@@ -503,22 +502,21 @@ protected function recordSaveAttendee(Vtiger_Record_Model $record, Sabre\VObject
...
 		$timeFormated = $time->format('Y-m-d H:i:s');
-		$db = \App\Db::getInstance();
-		$dbCommand = $db->createCommand();
+		$dbCommand = \App\Db::getInstance()->createCommand();
 		$attendees = $component->select('ATTENDEE');
 		foreach ($attendees as &$attendee) {
-			$value = ltrim($attendee->getValue(), 'mailto:');
-			if ($attendee['ROLE']->getValue() === 'CHAIR') {
-				$users = App\Fields\Email::findCrmidByEmail($value, ['Users']);
+			$value = preg_replace('/^mailto\:/i', '', $attendee->getValue());
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CalDAV.php]: ?

File: modules/API/models/CalDAV.php [hunk 13]
```diff
@@ -503,22 +502,21 @@ protected function recordSaveAttendee(Vtiger_Record_Model $record, Sabre\VObject
...
+			if ('CHAIR' === $attendee['ROLE']->getValue()) {
+				$users = \App\Fields\Email::findCrmidByEmail($value, ['Users']);
 				if (!empty($users)) {
 					continue;
 				}
 			}
 			$crmid = 0;
-			$records = App\Fields\Email::findCrmidByEmail($value, array_keys(array_merge(\App\ModuleHierarchy::getModulesByLevel(), \App\ModuleHierarchy::getModulesByLevel(3))));
+			$records = \App\Fields\Email::findCrmidByEmail($value, array_keys(array_merge(\App\ModuleHierarchy::getModulesByLevel(), \App\ModuleHierarchy::getModulesByLevel(3))));
```
Comment (TomaszQr) [inline modules/API/models/CalDAV.php]: \App\ModuleHierarchy::getModulesByLevel(4) ?

File: modules/Calendar/models/Record.php [hunk 14]
```diff
@@ -398,27 +394,40 @@ public function getRecordRelatedListViewLinksLeftSide(Vtiger_RelationListView_Mo
...
+	 * Get invitations with CRM metadata.
 	 *
-	 * @return array
+	 * @return void
 	 */
 	public function getInvities()
 	{
-		return (new \App\Db\Query())->from('u_#__activity_invitation')->where(['activityid' => (int) $this->getId()])->all();
+		return (new \App\Db\Query())
+			->select([
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/models/Record.php]: ?

**Pull request merged**